libevmasm/ThirdOptimizerTesting.cpp

```diff
@@ -0,0 +1,105 @@
+#include <libevmasm/ThirdOptimizerTesting.h>
+#include "ThirdOptimizerTesting.h"
+#include <algorithm>
+
+
+bool dev::solidity::Pattern::matches(std::vector<AssemblyItem> _items) const
```

Comment:
The other two implementations of `matches` receive a single expression, so this would correspond to a `std::iterator` or an offset inside the list .This means that the first loop below would be outside of this function in the optimizer stage and not in the pattern. But if this is easier for you to work with for now, it's of course fine.